task-sdk/src/airflow/sdk/_vendor/.gitignore

```
@@ -0,0 +1 @@
+*/*.py
```

```suggestion
*
```

Should not we ignore **everything** here ? Some of the vendored code might have resources.

---

We should also exclude (!)the __init__.py

---

And speaking of "_vendor" name - I think we should not use`_vendor` here as a name - even if we are using `vendoring` tool. Can we configure (mock?) the vendoring tool to use something different (_shared) sounds like a way better name). The `vendor` meaning is very tightly connected with `3rd party` but here we are `vendoring-in` our own code so we are really `1st party` here.

---

It will look better in the imports as well - reflecting it's real use, and if we go for functionality split that will be way more logical: 

```python
from airflow.sdk._shared.logging import logger
```

---

Yeah it's already a config setting in pyproject.toml

```toml
[tool.vendoring]
destination = "src/airflow/_vendor/"
requirements = "src/airflow/_vendor/vendor.txt"
namespace = "airflow._vendor"
```

---

We do have a quirk we have to avoid here -- if we have `shared/logging/src/logging/__init__.py` this would end up conflicting with the stdlib in dev using this approach. Perhaps another reason to suggest relative imports.

I guess the main issue with relative imports is are we okay with no imports between shared libs? For example, it might be nice to be able to use the timezone/date parsing stuff in the logging one, but I don't think we can do that when symlinking?

---

> I guess the main issue with relative imports is are we okay with no imports between shared libs? For example, it might be nice to be able to use the timezone/date parsing stuff in the logging one, but I don't think we can do that when symlinking?

I think (that was also my cyclomatic complexity arguments) we should avoid it as much as possible - because then it can very easily lead to circular dependencies (config -> logging - > config  that we currently have is a classic example of it). Most of this can be solved by injecting stuff at the higher level - for example during explicit initialization. It makes the "acyclic tree" hierarchy of dependencies between such shared code, which is precisely what low cyclomatic complexity is about. 

So in a way - adding "friction" for the shared code to use another shared code is likely a good thing. You should **not** be able to use the shared code from another module to easily. But also - it's not impossible in case of symlinks - you can have symlinks from one shared library to the other. That will work exactly the same way as symlinking shared library in the main code. And I think it's the right "amount of friction" - because you not only have to import stuff from the other library, but have to make symlink and it will - much more obviously - raise eyebrows during review - and cases like "logging uses config and config uses logging" is somethign that will be definitely "no go" as it will "visibly" create circular dependencies.